024-cyrus-sasl-2.1.21-sizes.patch

Taken from upstream, should be staple in next official release.
diff -r 190d1c9c1977 -r 6675a2cd7300 configure.ac
--- a/configure.ac	Thu Feb 02 08:30:42 2023 -0800
+++ b/configure.ac	Thu Feb 02 08:34:32 2023 -0800
@@ -1232,6 +1232,10 @@
 AC_HEADER_DIRENT
 AC_HEADER_SYS_WAIT
 AC_CHECK_HEADERS(crypt.h des.h dlfcn.h fcntl.h limits.h malloc.h paths.h strings.h sys/file.h sys/time.h syslog.h unistd.h inttypes.h sys/uio.h sys/param.h sysexits.h stdarg.h varargs.h krb5.h)
+AC_CHECK_TYPES([long long, int8_t, uint8_t, int16_t, uint16_t, int32_t, uint32_t, int64_t, uint64_t],,,[
+#ifdef HAVE_INTTYPES_H
+#include <inttypes.h>
+#endif])
 
 IPv6_CHECK_SS_FAMILY()
 IPv6_CHECK_SA_LEN()
diff -r 190d1c9c1977 -r 6675a2cd7300 include/makemd5.c
--- a/include/makemd5.c	Thu Feb 02 08:30:42 2023 -0800
+++ b/include/makemd5.c	Thu Feb 02 08:34:32 2023 -0800
@@ -82,12 +82,19 @@
  */
 
 
+#ifdef HAVE_CONFIG_H
+#include "../config.h"
+#endif
 
 #include <stdio.h>
 #include <string.h>
 #include <stdlib.h>
 #include <ctype.h>
 
+#ifdef HAVE_INTTYPES_H
+#include <inttypes.h>
+#endif
+
 
 static void
 my_strupr(char *s)
@@ -122,6 +129,18 @@
 static void
 try_signed(FILE *f, int len)
 {
+#ifdef HAVE_INT8_T
+    BITSIZE(int8_t);
+#endif
+#ifdef HAVE_INT16_T
+    BITSIZE(int16_t);
+#endif
+#ifdef HAVE_INT32_T
+    BITSIZE(int32_t);
+#endif
+#ifdef HAVE_INT64_T
+    BITSIZE(int64_t);
+#endif
     BITSIZE(signed char);
     BITSIZE(short);
     BITSIZE(int);
@@ -135,6 +154,18 @@
 static void
 try_unsigned(FILE *f, int len)
 {
+#ifdef HAVE_UINT8_T
+    BITSIZE(uint8_t);
+#endif
+#ifdef HAVE_UINT16_T
+    BITSIZE(uint16_t);
+#endif
+#ifdef HAVE_UINT32_T
+    BITSIZE(uint32_t);
+#endif
+#ifdef HAVE_UINT64_T
+    BITSIZE(uint64_t);
+#endif
     BITSIZE(unsigned char);
     BITSIZE(unsigned short);
     BITSIZE(unsigned int);
@@ -165,6 +196,11 @@
 	  "/* POINTER defines a generic pointer type */\n"
 	  "typedef unsigned char *POINTER;\n"
 	  "\n"
+#ifdef HAVE_INTTYPES_H
+	  "/* We try to define integer types for our use */\n"
+	  "#include <inttypes.h>\n"
+	  "\n"
+#endif
 	  );
   return 1;
 }
@@ -214,31 +250,15 @@
 
   print_pre(f);
 
-#ifndef HAVE_INT8_T
     try_signed (f, 8);
-#endif /* HAVE_INT8_T */
-#ifndef HAVE_INT16_T
     try_signed (f, 16);
-#endif /* HAVE_INT16_T */
-#ifndef HAVE_INT32_T
     try_signed (f, 32);
-#endif /* HAVE_INT32_T */
-#ifndef HAVE_INT64_T
     try_signed (f, 64);
-#endif /* HAVE_INT64_T */
 
-#ifndef HAVE_U_INT8_T
     try_unsigned (f, 8);
-#endif /* HAVE_INT8_T */
-#ifndef HAVE_U_INT16_T
     try_unsigned (f, 16);
-#endif /* HAVE_U_INT16_T */
-#ifndef HAVE_U_INT32_T
     try_unsigned (f, 32);
-#endif /* HAVE_U_INT32_T */
-#ifndef HAVE_U_INT64_T
     try_unsigned (f, 64);
-#endif /* HAVE_U_INT64_T */
 
     print_post(f);
   
